### PR TITLE
Kobel kostyantyn

### PR DESCRIPTION
**Pull request by kolyasalubov**

### CODE REVIEW

File: HW10/kkobel/ball_superr_ball_class.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+    def __init__(self, type='regular') -> None:
+        self.ball_of_type = type
+
+    def ball_type(self):
+        return self.ball_of_type
+
+ball1 = Ball()
+ball2 = Ball('super')
+print(ball1.ball_type())
+print(ball2.ball_type())
```
Comment (kolyasalubov) [inline HW10/kkobel/ball_superr_ball_class.py:15]: good job

File: HW10/kkobel/classy_classes.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+print(person1.info)
+
+person2 = Person("Matt",25)
+print(person2.info)
+
+person3 = Person("Alex",57)
+print(person3.info)
+
+person4 = Person("Cam",39)
+print(person4.info)
```
Comment (kolyasalubov) [inline HW10/kkobel/classy_classes.py:25]: good job

File: HW10/kkobel/double_char.py [hunk 3]
```diff
@@ -0,0 +1,7 @@
+def double_char(string):
+    return "".join(element*2 for element in string)
+
+
+print(f"Result: {double_char('String')}")
+print(f"Result: {double_char('Hello World')}")
+print(f"Result: {double_char('12345!')}")
```
Comment (kolyasalubov) [inline HW10/kkobel/double_char.py:7]: good job

File: HW10/kkobel/fix_the_loop.py [hunk 4]
```diff
@@ -0,0 +1,7 @@
+def list_animals(animals):
+    list = ''
+    for element in range(len(animals)):
+        list += str(element + 1) + '. ' + animals[element] + '\n'
+    return list
+
+print(f"Result of list animals: \n{list_animals([ 'dog', 'cat', 'elephant' ])}")
```
Comment (kolyasalubov) [inline HW10/kkobel/fix_the_loop.py:7]: good job

File: HW10/kkobel/ghost.py [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+        self.color = random.choice(colors)    
+
+    def getting_random_color(self):
+        return self.color           
+
+ghost1 = Ghost()
+print(f"Random color first object is: {ghost1.getting_random_color()}")
+
+ghost2 = Ghost()
+print(f"Random color second object is: {ghost2.getting_random_color()}")
```
Comment (kolyasalubov) [inline HW10/kkobel/ghost.py:27]: good job

File: HW10/kkobel/summation.py [hunk 6]
```diff
@@ -0,0 +1,10 @@
+def summation(num):
+    '''
+    Write a program that finds the summation of every number from 1 to num. 
+    The number will always be a positive integer greater than 0.
+    '''
+    return sum(element for element in range(1,num+1))
+
+print(f"Suma of numbers are: {summation(2)}")
+print(f"Suma of numbers are: {summation(4)}")
+print(f"Suma of numbers are: {summation(10)}")
```
Comment (kolyasalubov) [inline HW10/kkobel/summation.py:10]: good job

File: HW10/kkobel/task1.py [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+
+    def square_of_rectangle(self):
+        first_side, second_side, third_side, fourth_side = self.sides
+        square_of_rectangle = first_side**2 + second_side**2 + third_side**2 + fourth_side**2
+        print(f"Square of Rectangle is: {square_of_rectangle}") 
+
+rectangle1 = Rectangle()
+rectangle1.input_sides()
+rectangle1.dispSides()
+rectangle1.square_of_rectangle()
```
Comment (kolyasalubov) [inline HW10/kkobel/task1.py:31]: good job

File: HW10/kkobel/task2.py [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+    @staticmethod
+    def static_of_method():
+        return 'Good news for starting some new!'
+
+human1 = Human('Andrii')
+print(human1.greeting())
+print(human1.class_of_method())
+print(Human.class_of_method())  
+print(human1.static_of_method())
+
```
Comment (kolyasalubov) [inline HW10/kkobel/task2.py:30]: good job

File: HW10/kkobel/task3.py [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+print('Module of Class is:', Employee.__module__)
+print('Doc string of Class is:', Employee.__doc__)
+
+print('Module of instance is:', employee1.__module__)
+print('Doc string of instance is:', employee1.__doc__)
+print("Dict of instance is:", employee1.__dict__)
+
+print(employee1.info_of_employee())
+print(employee2.info_of_employee())
+print(employee3.info_of_employee())
```
Comment (kolyasalubov) [inline HW10/kkobel/task3.py:49]: good job

File: HW3/kkobel/task1.py [hunk 10]
```diff
@@ -21,9 +21,9 @@
...
 '''
-find_better_in_string = set_rules_of_Zen.find('better')
-find_never_in_string = set_rules_of_Zen.find('never')
-find_is_in_string = set_rules_of_Zen.find('is')
+find_better_in_string = set_rules_of_Zen.count('better')
+find_never_in_string = set_rules_of_Zen.count('never')
+find_is_in_string = set_rules_of_Zen.count('is')
 
 print(f"Count of better are: {find_better_in_string}")
 print(f"Count of never are: {find_never_in_string}")
```
Comment (kolyasalubov) [inline HW3/kkobel/task1.py:29]: good job

File: HW7/kkobel/task3.py [hunk 11]
```diff
@@ -2,11 +2,15 @@ def count_numbers_in_string(string):
...
+        if element in result_of_dictionary:
+            continue
+        else:
+            result_of_dictionary.update({str(element): string.count(element)})
+    return result_of_dictionary   
+
+print("Your count of string is:", count_numbers_in_string('Good days'))
+print("Your count of string is:", count_numbers_in_string('1005500'))
+print("Your count of string is:", count_numbers_in_string('$sabkjdnl'))
+print("Your count of input string is:", count_numbers_in_string(input("Enter your string: ")))
```
Comment (kolyasalubov) [inline HW7/kkobel/task3.py:16]: good job

File: HW8/kkobel/areas_of_figures.py [hunk 12]
```diff
@@ -0,0 +1,15 @@
...
+    return area_of_rectangle
+
+def triangle(side_a: int, height: int):
+    area_of_triangle = side_a*height*0.5
+    return area_of_triangle
+
+def circle(radius: int):
+    area_of_circle = math.pi*radius**2
+    change_to_two_decimal = '%.2f' % area_of_circle
+    return change_to_two_decimal
```
Comment (kolyasalubov) [inline HW8/kkobel/areas_of_figures.py:15]: good job

File: HW8/kkobel/convert_bool_values_to_string.py [hunk 13]
```diff
@@ -0,0 +1,13 @@
...
+
+def bool_to_word(boolean):
+    return "Yes" if boolean else "No"
+
+print(bool_to_word(2+2))
+print(bool_to_word(True))
+print(bool_to_word(12 > 0))
+print(bool_to_word(False))
+print(bool_to_word(0))
+print(bool_to_word(""))
```
Comment (kolyasalubov) [inline HW8/kkobel/convert_bool_values_to_string.py:13]: good job

File: HW8/kkobel/counting_sheep.py [hunk 14]
```diff
@@ -0,0 +1,21 @@
...
+def count_sheeps(sheep):
+    for item in sheep:
+        if item == True:
+            result_of_count = sheep.count(item)
+        else:
+            result = sheep.count(item)
+    print(f"Result of amount False or null or undefined are: {result}")
+    return result_of_count    
+
+print(f"Result of amount sheeps are: {count_sheeps(sheep)}")
```
Comment (kolyasalubov) [inline HW8/kkobel/counting_sheep.py:21]: good job

File: HW8/kkobel/is_this_my_tail.py [hunk 15]
```diff
@@ -0,0 +1,18 @@
...
+    sub = body.endswith(f"{tail}")
+    return True if sub  else  False
+
+def correct_tail(body, tail):
+    sub = body[(len(body)- len(tail))]
+    return True if sub == tail else False
+
+print(f"Result of string: {correct_tail('Fox', 'x')}")
+print(f"Result of string: {correct_tail('Elephant', 'x')}")
+print(f"Result of string: {correct_tail('Mouse', 'e')}")
```
Comment (kolyasalubov) [inline HW8/kkobel/is_this_my_tail.py:18]: good job

File: HW8/kkobel/multiplies_of_3_or_5.py [hunk 16]
```diff
@@ -0,0 +1,14 @@
...
+    list_numbers_division_by_3_or_5 = []
+    for item in range(1, user_input_number):
+        if item %3 == 0 or item %5 == 0:
+            sum_of_numbers += item
+            list_numbers_division_by_3_or_5.append(item)
+        elif item < 0:
+            return 0    
+    return list_numbers_division_by_3_or_5, sum_of_numbers              
+
+print(f"Result suma of numbers that division by 3 or 5: {solution(user_input_number)}")
```
Comment (kolyasalubov) [inline HW8/kkobel/multiplies_of_3_or_5.py:14]: good job

File: HW8/kkobel/playing_banjo.py [hunk 17]
```diff
@@ -0,0 +1,13 @@
...
+
+The function takes a name as its only argument, and returns one of the following strings:
+'''
+
+
+def are_you_playing_banjo(name: str):
+    return name + ' plays banjo' if name[0].lower() == 'r' or name[0].upper() == 'R'  else name + ' does not play banjo'
+
+user_input = are_you_playing_banjo(input('Enter your name: '))
+print(user_input)
```
Comment (kolyasalubov) [inline HW8/kkobel/playing_banjo.py:13]: good job

File: HW8/kkobel/reverse_list.py [hunk 18]
```diff
@@ -0,0 +1,6 @@
+l = [1,2,3,4]
+def reversing_list(l):
+    result = list(reversed(l))
+    return result
+
+print(f"Reversing list is: {reversing_list(l)}")
```
Comment (kolyasalubov) [inline HW8/kkobel/reverse_list.py:6]: good job

File: HW8/kkobel/task1.py [hunk 19]
```diff
@@ -0,0 +1,16 @@
...
+    elif user_input == 'triangle':
+        print(f"Area of Triangle is: {areas_of_figures.triangle(int(input('Enter side_a: ')), int(input('Enter heights: ')))}")
+    elif user_input == 'circle':
+        print(f"Area of Circle is: {areas_of_figures.circle(int(input('Enter radius: ')))}")
+    else:
+        print('You entered not correct function')
+        print('Try again, please')
+        user_input = calculating_areas(input('Enter function (rectangle, triangle, circle) to calculate areas : '))        
+
+user_input = calculating_areas(input('Enter function (rectangle, triangle, circle) to calculate areas : '))
```
Comment (kolyasalubov) [inline HW8/kkobel/task1.py:16]: good job

File: HW8/kkobel/task2.py [hunk 20]
```diff
@@ -0,0 +1,46 @@
...
+    Maximum length 16 characters.
+    '''
+    validate_of_lower_letters = re.findall(("[a-z]") , password)
+    validate_of_upper_letters = re.findall(("[A-Z]") , password)
+    validate_of_numbers = re.findall(("[0-9]") , password)
+    validate_of_symbols = re.findall(("[$#@]") , password)
+
+    flag = True
+    while flag:
+        if len(password) < 6:
```
Comment (kolyasalubov) [inline HW8/kkobel/task2.py:21]: you can use the condition  16< len(password) < 6:

File: HW8/kkobel/task_of_pygame.py [hunk 21]
```diff
@@ -0,0 +1,58 @@
...
+
+
+    gameDisplay.fill(BLACK_COLOR) 
+
+    pygame.draw.rect(gameDisplay, (255,0,0), [COORD_X, 
+                                              COORD_Y, 
+                                              WIDTH_RECTANGLE, 
+                                              HEIGHT_RECTANGLE])
+    pygame.display.update()
+    clock.tick(FPS)
```
Comment (kolyasalubov) [inline HW8/kkobel/task_of_pygame.py:58]: good job

File: HW8/kkobel/weather_of_pyowm.py [hunk 22]
```diff
@@ -0,0 +1,18 @@
...
+observation = manager_of_weather.weather_at_place('Lviv,Ukraine')
+location_of_weather = observation.weather
+
+print(f"Detailed status is: {location_of_weather.detailed_status}")
+print(f"Wind is: {location_of_weather.wind()}")
+print(f"Humidity is: {location_of_weather.humidity}")
+print(f"Temperature is: {location_of_weather.temperature('celsius')}")
+print(f"Rain is: {location_of_weather.rain}")
+print(f"Heat index is: {location_of_weather.heat_index}")
+print(f"Clouds are: {location_of_weather.clouds}")
```
Comment (kolyasalubov) [inline HW8/kkobel/weather_of_pyowm.py:18]: good job

File: HW8/kkobel/wiil_you_make_it.py [hunk 23]
```diff
@@ -0,0 +1,14 @@
...
+'''
+def zero_fuel(distance_to_pump, mpg, fuel_left):
+    result_of_distance = distance_to_pump / mpg
+    return True if fuel_left > result_of_distance or fuel_left >= result_of_distance else False 
+
+
+print(f"Result of your pump is: {zero_fuel(50,25,1)}")
+print(f"Result of your pump is: {zero_fuel(50,25,2)}")
+print(f"Result of your pump is: {zero_fuel(30,25,1)}")
+print(f"Result of your pump is: {zero_fuel(30,25,1.2)}")
```
Comment (kolyasalubov) [inline HW8/kkobel/wiil_you_make_it.py:14]: good job

File: HW8/kkobel/works_with_random_numbers.py [hunk 24]
```diff
@@ -0,0 +1,16 @@
...
+    if user_input == result_of_random:
+        print(f"Your number is: {user_input}")
+        print('Congratilations you win your prize!!!')
+        break
+    elif user_input > result_of_random:    
+        print(f"Your number {user_input} is bigger than Our Random Number")
+        user_input = int(input('Enter integer number: '))
+    elif user_input < result_of_random:
+        print(f"Your number {user_input} is less than Our Random Number")
+        user_input = int(input('Enter integer number: '))
```
Comment (kolyasalubov) [inline HW8/kkobel/works_with_random_numbers.py:16]: good job

File: HW8/kkobel/task2.py (hunk 20 above)
Comment (KostyaKobel) [reply HW8/kkobel/task2.py:21]: Yes sure. But I wanted to show you a good print of the information for each of the conditions.

**Pull request merged**